src/routes/docs/tutorials/nextjs/step-7/+page.markdoc

```diff
@@ -0,0 +1,410 @@
+---
+layout: tutorial
+title: Edit and delete expenses
+description: Edit and delete expenses from expense list
+step: 7
+---
+
+In the expense card we have added two buttons to edit and delete the expense. We will add the functionality to these buttons in this step.
+
+# Edit expense {% #edit-expense %}
+
+First create a API to edit the expense. We will create `PUT` api in the same `/src/app/api/expense/route.js` file. We will use the `updateDocument` method of the `database` service to update the expense document.
+
+```js
+import { ID } from "appwrite";
+import { headers } from "next/headers";
+import { NextResponse } from "next/server";
+
+import {
+  EXPENSE_COLLECTION_ID,
+  EXPENSE_DATABASE_ID,
+  account,
+  database,
+} from "@/config/appwrite";
+
+// Post - create expense
+
+// Put - update expense
+export async function PUT(req) {
+  const body = await req.json();
+  const { title, amount, type, docId } = body;
+  if (!title || !amount || !type || !docId) {
+    return new NextResponse("Missing fields", { status: 401 });
+  }
+
+  const headersList = headers();
+  const jwt = headersList.get("jwt");
+
+  if (!jwt) {
+    return new NextResponse("Unauthorized", { status: 401 });
+  }
+
+  account.client.setJWT(jwt);
+
+  try {
+    await database.updateDocument(
+      EXPENSE_DATABASE_ID,
+      EXPENSE_COLLECTION_ID,
+      docId,
+      {
+        title: title,
+        amount: amount,
+        type: type,
+      }
+    );
+
+    return new NextResponse("Updated successfully", { status: 200 });
+  } catch (error) {
+    return new NextResponse("Internal server error", {
+      status: 500,
+      error: error.message,
+    });
+  }
+}
+```
+
+When the user clicks on the edit button, we will show the expense popup form with the expense details filled in. The user can then edit the expense and save it.
+
+We will add a click event handler to the edit button in the `ExpenseCard` component.
```

Comment:
Where did the expense card component come from? It's not in the tutorial?

Reply:
My mistake, It is ExpenseList